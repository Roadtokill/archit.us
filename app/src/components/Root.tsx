import { css } from "linaria";
import React from "react";
import { Provider } from "react-redux";

import ColorModeProvider from "@app/components/ColorModeProvider";
import NotificationPane from "@app/components/NotificationPane";
import { Link } from "@app/components/Router";
import Store from "@app/store";
import { injectColorGlobals } from "@app/theme/color";
import { AutoLinkContext } from "@architus/facade/components/AutoLink";

// Import the global CSS rules from Linaria
import "@architus/facade/theme/globals";
import "@app/layout";

// Inject the Gatsby router link to auto links
const linkContext: AutoLinkContext = { link: Link };

// Import the global app colors
export const rootGlobal = css`
  :global() {
    ${injectColorGlobals()}
  }
`;

/**
 * Represents the root component of the application,
 * which remains mounted across navigations
 */
const Root: React.FC = ({ children }) => (
  <AutoLinkContext.Provider value={linkContext}>
    <ColorModeProvider>
      <Provider store={Store}>
        <NotificationPane />
        {children}
      </Provider>
    </ColorModeProvider>
  </AutoLinkContext.Provider>
);

export default Root;
