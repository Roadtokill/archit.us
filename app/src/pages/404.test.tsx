import { describe, it, expect } from "@jest/globals";
import React from "react";
import renderer from "react-test-renderer";

import "@app/test/watch-media";
import NotFoundPage from "./404";
import Root from "@app/components/Root";

describe("<NotFoundPage />", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Root>
          <NotFoundPage />
        </Root>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
