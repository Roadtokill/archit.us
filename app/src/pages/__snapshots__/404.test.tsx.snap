// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFoundPage /> renders correctly 1`] = `
Array [
  <div
    className="notification-pane"
  />,
  <div
    className=" css-18nmgyy"
    style={Object {}}
  >
    <nav
      className="navbar navbar-expand-md navbar-dark sticky-top"
    >
      <div
        className="container"
      >
        <div
          className="css-1e1rowh"
          display="flex"
        >
          <a
            aria-current="page"
            className="nav-link brand"
            href="/"
            onClick={[Function]}
          >
            <svg
              height="40px"
            >
              __svg__
            </svg>
          </a>
          <div
            onClick={[Function]}
          >
            <div
              className="css-mo8bn3"
              display={
                Object {
                  "lg": "inline-block",
                  "xs": "none",
                }
              }
            >
              LOCAL
            </div>
          </div>
        </div>
        <ul
          className="navbar-nav mr-auto"
        />
        <div
          className="header-children"
        >
          <div
            className="css-l1w1vw"
          >
            <div
              className="css-15bd3x9"
              style={
                Object {
                  "MozTransition": "opacity 0.25s",
                  "MozUserSelect": "none",
                  "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                  "WebkitTransition": "opacity 0.25s",
                  "WebkitUserSelect": "none",
                  "borderRadius": 14,
                  "direction": "ltr",
                  "display": "inline-block",
                  "msUserSelect": "none",
                  "opacity": 1,
                  "position": "relative",
                  "textAlign": "left",
                  "touchAction": "none",
                  "transition": "opacity 0.25s",
                  "userSelect": "none",
                }
              }
            >
              <div
                className="react-switch-bg"
                onClick={[Function]}
                onMouseDown={[Function]}
                style={
                  Object {
                    "MozTransition": "background 0.25s",
                    "WebkitTransition": "background 0.25s",
                    "background": "#ffffff",
                    "borderRadius": 14,
                    "cursor": "pointer",
                    "height": 28,
                    "margin": 0,
                    "position": "relative",
                    "transition": "background 0.25s",
                    "width": 56,
                  }
                }
              >
                <div
                  style={
                    Object {
                      "MozTransition": "opacity 0.25s",
                      "WebkitTransition": "opacity 0.25s",
                      "height": 28,
                      "opacity": 0,
                      "pointerEvents": "none",
                      "position": "relative",
                      "transition": "opacity 0.25s",
                      "width": 30,
                    }
                  }
                >
                  <span
                    className="icon css-17mj8ar"
                    color="light"
                  >
                    <svg
                      aria-hidden="true"
                      className="svg-inline--fa fa-moon fa-w-16 "
                      data-icon="moon"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      style={Object {}}
                      viewBox="0 0 512 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M283.211 512c78.962 0 151.079-35.925 198.857-94.792 7.068-8.708-.639-21.43-11.562-19.35-124.203 23.654-238.262-71.576-238.262-196.954 0-72.222 38.662-138.635 101.498-174.394 9.686-5.512 7.25-20.197-3.756-22.23A258.156 258.156 0 0 0 283.211 0c-141.309 0-256 114.511-256 256 0 141.309 114.511 256 256 256z"
                        fill="currentColor"
                        style={Object {}}
                      />
                    </svg>
                  </span>
                </div>
                <div
                  style={
                    Object {
                      "MozTransition": "opacity 0.25s",
                      "WebkitTransition": "opacity 0.25s",
                      "height": 28,
                      "opacity": 1,
                      "pointerEvents": "none",
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "transition": "opacity 0.25s",
                      "width": 30,
                    }
                  }
                >
                  <span
                    className="icon css-d6hmy4"
                    color="dark"
                  >
                    <svg
                      aria-hidden="true"
                      className="svg-inline--fa fa-sun fa-w-16 "
                      data-icon="sun"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      style={Object {}}
                      viewBox="0 0 512 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M256 160c-52.9 0-96 43.1-96 96s43.1 96 96 96 96-43.1 96-96-43.1-96-96-96zm246.4 80.5l-94.7-47.3 33.5-100.4c4.5-13.6-8.4-26.5-21.9-21.9l-100.4 33.5-47.4-94.8c-6.4-12.8-24.6-12.8-31 0l-47.3 94.7L92.7 70.8c-13.6-4.5-26.5 8.4-21.9 21.9l33.5 100.4-94.7 47.4c-12.8 6.4-12.8 24.6 0 31l94.7 47.3-33.5 100.5c-4.5 13.6 8.4 26.5 21.9 21.9l100.4-33.5 47.3 94.7c6.4 12.8 24.6 12.8 31 0l47.3-94.7 100.4 33.5c13.6 4.5 26.5-8.4 21.9-21.9l-33.5-100.4 94.7-47.3c13-6.5 13-24.7.2-31.1zm-155.9 106c-49.9 49.9-131.1 49.9-181 0-49.9-49.9-49.9-131.1 0-181 49.9-49.9 131.1-49.9 181 0 49.9 49.9 49.9 131.1 0 181z"
                        fill="currentColor"
                        style={Object {}}
                      />
                    </svg>
                  </span>
                </div>
              </div>
              <div
                className="react-switch-handle"
                onClick={[Function]}
                onMouseDown={[Function]}
                onTouchCancel={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                style={
                  Object {
                    "MozTransition": "background-color 0.25s, transform 0.25s, box-shadow 0.15s",
                    "WebkitTransition": "background-color 0.25s, transform 0.25s, box-shadow 0.15s",
                    "background": "#e8eaeb",
                    "border": 0,
                    "borderRadius": "50%",
                    "boxShadow": "0px 1px 5px rgba(0, 0, 0, 0.6)",
                    "cursor": "pointer",
                    "display": "inline-block",
                    "height": 26,
                    "outline": 0,
                    "position": "absolute",
                    "top": 1,
                    "transform": "translateX(1px)",
                    "transition": "background-color 0.25s, transform 0.25s, box-shadow 0.15s",
                    "width": 26,
                  }
                }
              />
              <input
                aria-label="Dark mode switch"
                checked={false}
                disabled={false}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyUp={[Function]}
                role="switch"
                style={
                  Object {
                    "border": 0,
                    "clip": "rect(0 0 0 0)",
                    "height": 1,
                    "margin": -1,
                    "overflow": "hidden",
                    "padding": 0,
                    "position": "absolute",
                    "width": 1,
                  }
                }
                type="checkbox"
              />
            </div>
          </div>
          <a
            className="nav-link"
            href="https://api.develop.archit.us/session/login?return=http%3A%2F%2Flocalhost%2Fapp"
          >
            Sign In
            <span
              className="icon css-hy30ic"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-external-link-alt fa-w-16 "
                data-icon="external-link-alt"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </span>
          </a>
        </div>
      </div>
    </nav>
    <div
      className="Layout_l14wiqln"
    >
      <div
        className="Content_cj6xfbl"
      >
        <h2
          className="PageTitle_p1vzh75s"
        >
          404
        </h2>
        <h3
          className="Subtitle_si8t7dg"
        >
          Page not found
        </h3>
        <p
          className="Paragraph_p1v1i8tf"
        >
          The page you're looking for doesn't exist or was removed.
        </p>
        <p
          className="Paragraph_p1v1i8tf"
        >
          If you feel this is an error, please file a new issue
           
          <a
            href="https://github.com/architus/archit.us/issues/new"
          >
            on GitHub
            <span
              className="icon css-hy30ic"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-external-link-alt fa-w-16 "
                data-icon="external-link-alt"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </span>
          </a>
        </p>
      </div>
    </div>
  </div>,
]
`;
